### PR TITLE
Concrete ident for coq

### PR DESCRIPTION
**Pull request by cmester0**
Move concrete ident modifications for Coq backend (#987) to separate PR.

### CODE REVIEW

**Comment by cmester0** [comment]
For Coq I cannot name tuple fields, 0,1,2,.., furthermore fields need to be unique, so we should use the path in the name of the fields. This naming also has to be consistent when used elsewhere, so I changing this in concrete ident.

File: test-harness/src/snapshots/toolchain__reordering into-coq.snap [hunk 1]
```diff
@@ -42,33 +42,33 @@ Import RecordSetNotations.
...
-| Foo_B.
+| t_Foo_A
+| t_Foo_B.
 Arguments t_Foo:clear implicits.
 Arguments t_Foo.
 
 Record t_Bar : Type :=
   {
-    0 : t_Foo;
+    t_Bar_0 : t_Foo;
```
Comment (cmester0) [inline test-harness/src/snapshots/toolchain__reordering into-coq.snap:52]: This is a solution to tuple structs.

File: test-harness/src/snapshots/toolchain__literals into-coq.snap [hunk 2]
```diff
@@ -47,13 +47,13 @@ Export Hax_lib (t_int).
 
 Record t_Foo : Type :=
   {
-    f_field : t_u8;
+    t_Foo_f_field : t_u8;
```
Comment (cmester0) [inline test-harness/src/snapshots/toolchain__literals into-coq.snap:50]: This is a solution to overlapping field names in structs.

Comment (W95Psp) [reply test-harness/src/snapshots/toolchain__literals into-coq.snap:50]: Oh, I see!
Then you need to improve the name policies API: the binding `enum_constructor_name_transform` is what you want, but you need to add an extra argument "struct_name" or something, just as I did for `enum_constructor_name_transform`.
Do you see what I mean?

Comment (cmester0) [reply test-harness/src/snapshots/toolchain__literals into-coq.snap:50]: Ahh, yes. Makes sense.

Comment (cmester0) [reply test-harness/src/snapshots/toolchain__literals into-coq.snap:50]: This will also fix the issue I had, with overwriting keywords (e.g. `end`).

Comment (W95Psp) [reply test-harness/src/snapshots/toolchain__literals into-coq.snap:50]: Great! :tada:

**Comment by cmester0** [comment]
Closed in favor of #1108, which has more features

**Pull request closed without merging**